src/event-tracker.ts

```diff
@@ -78,9 +82,16 @@ export class EventTracker {
 		}
 
 		properties = this.assignDefaultProperties(properties);
-		this.client.track(event, properties);
+		this.debouncedLogger(event)(properties);
 	}
 
+	private debouncedLogger = _.memoize((event) => {
```

Comment:
You can use `memoizee` with `primitive: true` for this - we already have memoizee installed in the supervisor and it's what we normally use for memoization

Reply:
Done - merging.